pkg/xray/provider.go

```diff
@@ -133,3 +146,24 @@ func sendUsageRepo(restyBase *resty.Client, terraformVersion string) (interface{
 	}
 	return nil, nil
 }
+
+func checkArtifactoryLicense(client *resty.Client) error {
```

Comment:
@alexhung didn't we discuss how we should make license check a thing you could switch off?

Reply:
@chb0github Yes, I haven't had time to work on it. We can implement it here but I prefer it as a separate PR.